packages/frontend-plugin-api/src/components/ExtensionBoundary.test.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect } from 'react';
+import { screen, waitFor } from '@testing-library/react';
+import {
+  MockAnalyticsApi,
+  MockConfigApi,
+  TestApiProvider,
+  renderWithEffects,
+} from '@backstage/test-utils';
+import { ExtensionBoundary } from './ExtensionBoundary';
+import {
+  Extension,
+  coreExtensionData,
+  createExtension,
+  createPlugin,
+} from '../wiring';
+import {
+  analyticsApiRef,
+  createRouteRef,
+  useAnalytics,
+} from '@backstage/core-plugin-api';
+import { createApp } from '@backstage/frontend-app-api';
+import { JsonObject } from '@backstage/types';
+
+function renderExtensionInTestApp(
```

Comment:
> [!NOTE]
> Couldn't customize the legacy app `ErrorBoundaryFallback ` component and it throws an error, so I'm using a custom renderer to test with the default app error component:

Test `ErrorBoundaryFallback`
https://github.com/backstage/backstage/blob/2e889f1fe9bffc81d72698357644d49225e66d49/packages/test-utils/src/testUtils/appWrappers.tsx#L66-L68

Default `ErrorBoundaryFallback`
https://github.com/backstage/backstage/blob/2e889f1fe9bffc81d72698357644d49225e66d49/packages/app-defaults/src/defaults/components.tsx#L57-L73

Reply:
Not relevant anymore since we are not using app components.